core/types/src/attestation_key.rs

```diff
@@ -0,0 +1,271 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+//! Attestation Key types
+
+use crate::{
+    attestation_key::QuoteSignature::{Linkable, UnLinkable},
+    new_type_accessors_impls,
+    report::{ExtendedProductId, FamilyId},
+    ConfigId, FfiError,
+};
+use mc_sgx_core_sys_types::{sgx_att_key_id_ext_t, sgx_ql_att_key_id_t, sgx_quote_sign_type_t};
+
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+pub enum MrSignerKeyHash {
+    Sha256([u8; 32]),
+    Sha384([u8; 48]),
+}
+
+#[derive(Debug, Default, Clone, Hash, PartialEq, Eq)]
+#[repr(transparent)]
+pub struct AlgorithmId(u32);
```

Comment:
If this corresponds to `sgx_ql_attestation_algorithm_id_t`, shouldn't we be restricting it from taking on values which are (currently) considered illegal? Currently it's unconstrained (and you even have it with a value of 9 in your test, which is above `SGX_QL_ALG_MAX`)

Reply:
Converted `Algorithm` into an enum, that has a `TryFrom<u32>`.  This should lay the groundwork for when the `mc-sgx-dcap-types` comes into play